cpp/src/parquet/arrow/schema.cc

```diff
@@ -681,6 +681,10 @@ Status ListToSchemaField(const GroupNode& group, LevelInfo current_levels,
       // List of primitive type
       RETURN_NOT_OK(
           NodeToSchemaField(*list_group.field(0), current_levels, ctx, out, child_field));
+    } else if (list_group.field_count() == 1 && list_group.field(0)->is_repeated()) {
```

Comment:
BTW, it looks like the HasStructListName is not correct for _tuple, as it only checks the name ends in `_tuple` and not that it is the top level list appended with `_tuple`.

Reply:
The rule 3 has following implications:
- The repeated field is a group with one field and is named array, whether or not LIST-annotated.
- The repeated field is a group with one field and uses the LIST-annotated group's name with _tuple appended.

Am I right? @emkornfield